app/assets/javascripts/controllers/ems_common/arbitration_profile_form_controller.js

```diff
@@ -0,0 +1,130 @@
+    };
+    $scope.formId    = arbitrationProfileFormId;
+    $scope.afterGet  = false;
+    $scope.model     = "arbitrationProfileModel";
+    ManageIQ.angular.scope = $scope;
+
+    if (arbitrationProfileFormId == 'new') {
+      $scope.newRecord = true;
+      $scope.arbitrationProfileModel.ems_id = emsId;
+      $scope.profileOptions($scope.arbitrationProfileModel.ems_id, $scope.arbitrationProfileModel.cloud_network_id);
+      $scope.modelCopy = angular.copy( $scope.arbitrationProfileModel );
+    } else {
+      var profileId = queryParam('show').toString();
+      arbitrationProfileDataFactory.getArbitrationProfileData(profileId).then (function(arbitrationProfileData) {
+        $scope.newRecord = false;
+        $scope.arbitrationProfileModel.name                 = arbitrationProfileData.name;
+        $scope.arbitrationProfileModel.description          = arbitrationProfileData.description;
+        $scope.arbitrationProfileModel.authentication_id    = convertToString(arbitrationProfileData.authentication_id);
+        $scope.arbitrationProfileModel.availability_zone_id = convertToString(arbitrationProfileData.availability_zone_id);
+        $scope.arbitrationProfileModel.cloud_network_id     = convertToString(arbitrationProfileData.cloud_network_id);
+        $scope.arbitrationProfileModel.cloud_subnet_id      = convertToString(arbitrationProfileData.cloud_subnet_id);
+        $scope.arbitrationProfileModel.flavor_id            = convertToString(arbitrationProfileData.flavor_id);
+        $scope.arbitrationProfileModel.security_group_id    = convertToString(arbitrationProfileData.security_group_id);
+
+        $scope.profileOptions($scope.arbitrationProfileModel.ems_id, $scope.arbitrationProfileModel.cloud_network_id);
+        $scope.modelCopy = angular.copy( $scope.arbitrationProfileModel );
+      });
+    }
+  };
+
+  $scope.cancelClicked = function() {
+    if ($scope.newRecord)
+      var msg = sprintf(__("Add of Arbitration Profile was cancelled by the user"));
+    else
+      var msg = sprintf(__("Edit of Arbitration Profile %s was cancelled by the user"), $scope.arbitrationProfileModel.description);
+    postService.cancelOperation(redirectUrl, msg);
+    $scope.angularForm.$setPristine(true);
+  };
+
+  $scope.resetClicked = function() {
+    $scope.arbitrationProfileModel = angular.copy( $scope.modelCopy );
+    $scope.profileOptions($scope.arbitrationProfileModel.ems_id, $scope.arbitrationProfileModel.cloud_network_id);
+    $scope.angularForm.$setUntouched(true);
+    $scope.angularForm.$setPristine(true);
+    miqService.miqFlash("warn", __("All changes have been reset"));
+  };
+
+  $scope.saveClicked = function() {
+    var successMsg = sprintf(__("Arbitration Profile %s was saved"), $scope.arbitrationProfileModel.name);
+    postService.saveRecord('/api/arbitration_profiles/' + arbitrationProfileFormId,
+      redirectUrl,
+      setProfileOptions(),
+      successMsg);
+    $scope.angularForm.$setPristine(true);
```

Comment:
I think we don't need this line.. (and same for add..) .. because, if it succeeds, we leave the screen anyway.. and if it fails, we don't want to mark the form as pristine.. 

(Also.. seems like marking as pristine should go hand in hand with overwriting modelCopy..)